src/main/java/frc/robot/component/CurrentTrigger.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.component;
+
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.units.Current;
+import edu.wpi.first.units.Measure;
+import edu.wpi.first.units.Time;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.util.MoPrefs.UnitPref;
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+/** Detects overcurrent on motor controllers, serves as limit detection */
+public class CurrentTrigger implements BooleanSupplier {
+    private final Supplier<Measure<Current>> current;
+    public final Trigger trigger;
+    public Measure<Current> currentThreshold;
+    public Measure<Time> triggerDuration;
+
+    private long triggerStartTimeMs = 0;
+
+    public CurrentTrigger(
```

Comment:
Not sure why this needs to be broken out into its own class, when the core logic is just a 4-line if-statement.

Refer to the ZeroIntakeCommand to see how I would do it.